crates/orchestrator/src/network/node.rs

```diff
@@ -37,6 +37,59 @@ pub struct NetworkNode {
     metrics_cache: Arc<RwLock<MetricMap>>,
 }
 
+/// Result of waiting for a certain number of log lines to appear.
+///
+/// Indicates whether the log line count condition was met within the timeout period.
+///
+/// # Variants
+/// - `TargetReached(count, duration_ms)` – The predicate condition was satisfied within the timeout.
+///     * `count`: The number of matching log lines at the time of satisfaction.
+///     * `duration_ms`: Time elapsed (in milliseconds) until the condition was met.
+/// - `TargetFailed(count)` – The condition was not met within the timeout.
+///     * `count`: The final number of matching log lines at timeout expiration.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum LogLineCount {
+    TargetReached(u32, u64),
```

Comment:
out of curiosity,  did you have a use-case that needs the amount of time taken to reach the target?

Reply:
Actually no :) I will remove it.